searx/searx
# [mod] upgrade pygments

## What does this PR do?

* upgrade pygments to the last version
* add `searx_data/update/update_pygments.py` to update the css styles of the oscar and simple themes.
    * the script creates a `pygments.less` file for each theme
    * `code.less` includes `pygments.less`
* oscar theme: fix the `.highlight` CSS class. In the master branch:
    * one usage hightlights the words from the user query (using bold font)
    * another usage highlights the source code. This PR renames this CSS class to `.highlight-code`, so the source code don't use a bold font (but perhaps it was intended ?)
* the pygments styles are nearly identical.

## Why is this change important?

The version 2.1.3 is 5 years old, and starts to create incompatibility problem. See https://github.com/searx/searx/issues/2204

Most probably it is possible to the old versions of the different dependencies, but it would be good to be able to upgrade easily.
Currently it is difficult because the pygments CSS are hard-coded into the simple and oscar themes:
* https://github.com/searx/searx/blob/master/searx/static/themes/simple/less/code.less
* https://github.com/searx/searx/blob/master/searx/static/themes/oscar/less/logicodev/code.less
* https://github.com/searx/searx/blob/master/searx/static/themes/oscar/less/pointhi/code.less

The ```.highlight *``` CSS are actually the result of a Python method call: ```HtmlFormatter(style=style).get_style_defs(cssclass)```.

This PR adds a script to easily update the themes ( utils/update-pygments.py ).

## How to test this PR locally?

With all the themes & style, search for ```!scc searx```.
Compare visually the result with a searx instance without this PR.
The results should be identical:
* mouse cursor on code, line number
* being able to select the code without the line number.

## Author's checklist



## Additional note

This PR commits the `pygments.less` files, so `searx_data/update/update_pygments.py` must be called after each upgrade of pygments.

Another PR can improve that:
* the `pygments.less` files are not commited (in .gitignore)
* there are generated each time when the themes are built.
* since the built css are commited too, it doesn't solve the issue.

## Related issues

Related to:
* #2204
* #2034
